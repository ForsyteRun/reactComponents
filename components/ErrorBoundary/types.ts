import { ErrorInfo, ReactNode } from 'react';

export type ErrorState = {
  error: Error | null;
  errorInfo: ErrorInfo | null;
};

export type ErrorProps = {
  children: ReactNode;
};
