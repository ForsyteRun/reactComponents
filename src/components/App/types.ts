import { IItem } from '../../types';

export type StateType = {
  items: IItem[];
  error: boolean;
  loading: boolean;
};
