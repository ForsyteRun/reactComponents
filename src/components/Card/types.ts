import { IItem } from '../../types';

export interface ICard {
  data: IItem;
  handleClick: (id: string) => void;
}
