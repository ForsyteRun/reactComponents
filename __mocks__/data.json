{
  "status": 200,
  "kind": "books#volumes",
  "totalItems": 1392,
  "items": [
    {
      "id": "pfaeBAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "8cm3DwAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "xcRiAAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "NmseEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "8ZeGEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "c-g2DwAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "0HkVEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "5_cwDwAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "xJOnEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "qer7ic8z03IC",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "aW6fEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "Y0yFEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "ma8VEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "6hzsWwz1aLAC",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "4ec6AQAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "37ieAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "v-ynEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "kTwpAAAAYAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "vjkEAQAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "Q1BpAAAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "cTZW0qg-uecC",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "FvXZAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "U-kaAAAAYAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "hd0mAQAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "SxcJAAAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },

    {
      "id": "M0sbAAAAYAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "f15qAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "xT7xAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "DnBPAAAAYAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "vXRHAAAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "gTUm9j-jLVAC",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "SJ9PAQAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "C9RKAQAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "s4FtAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "cbxNAQAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "E-hPAAAAYAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "NasbAQAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "EtIdAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "o9tqAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "3zWQAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    }
  ]
}
