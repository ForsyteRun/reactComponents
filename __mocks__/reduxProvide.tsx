import { ThunkMiddleware, configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { booksApi } from '../src/services/fetchData';
import paginationSlice from './../src/store/slices/pagination';
import searchSlice from './../src/store/slices/search';
import cardSlice from './../src/store/slices/card';
import loadingSlice from './../src/store/slices/loading';

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: {
        [booksApi.reducerPath]: booksApi.reducer,
        search: searchSlice,
        pagination: paginationSlice,
        card: cardSlice,
        loading: loadingSlice,
      },
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware().concat(booksApi.middleware as ThunkMiddleware),
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <BrowserRouter>
        <Provider store={store}>{children}</Provider>;
      </BrowserRouter>
    );
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
