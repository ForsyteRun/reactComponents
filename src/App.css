#root {
  max-width: 330px;
  margin: 0 auto;
  text-align: center;
}

a {
  text-decoration: none;
  cursor: pointer;
}

button {
  cursor: pointer;
}

.title {
  margin-bottom: 1rem;
}

form {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: #f5f0f0;
  border-radius: 0.51rem;
  box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
  padding: 1rem;
}

form > label {
  margin-bottom: 1rem;
}

form > label > input {
  border: 0.5px solid transparent;
  border-radius: 2px;
  padding: 0.2rem;
}

form > button {
  width: 100%;
  border-radius: 0.2rem;
  background: #6d2bd6;
  color: aliceblue;
  border: transparent;
  padding: 0.5rem 0.3rem;
  box-shadow: rgba(0, 0, 0, 0.773) 0.3rem 3px 0.4rem;
  transition: all 0.1s ease-out;
  cursor: pointer;
}

form > button:active {
  transform: translate(2px, 2px);
  transition: all 0.1s ease-out;
}

form > button:disabled {
  background: #4a494d;
  color: rgb(205, 209, 213);
  transform: none;
}

.upload {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-end;
  border-bottom: 1px solid #9999997c;
  padding: 0.5rem 0;
}

.upload > h5 {
  align-self: normal;
}

h5 {
  margin: 0;
  padding: 0;
  color: red;
  font-size: 0.6rem;
}

.inputContainer {
  height: 3rem;
}

.easyLevel {
  color: rgb(238, 58, 8);
}

.normalLevel {
  color: rgb(34, 0, 128);
}

.hardLevel {
  color: rgb(0, 128, 0);
}

.containerPasswordLevel {
  position: absolute;
  right: -3.5rem;
  top: 0;
  font-weight: 700;
}
