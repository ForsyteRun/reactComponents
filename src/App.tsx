// import { useState } from 'react';
import { useLoaderData } from 'react-router-dom';

import BooksProvider from './context/BooksProvider';
import SearchProvider from './context/SearchProvider';
import { Home } from './pages';
import { IFetchData, storageData } from './types';
import { getStorageData } from './utils';

const App = () => {
  const fetchInitData = useLoaderData() as IFetchData;

  const storageQuery = getStorageData(storageData.formValue);

  return (
    <SearchProvider value={storageQuery}>
      <BooksProvider value={fetchInitData.items}>
        <Home />
      </BooksProvider>
    </SearchProvider>
  );
};

export default App;
