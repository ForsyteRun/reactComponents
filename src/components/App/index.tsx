import React from 'react';
import { StateType } from '../../types';
import s from './App.module.css';
import ListItems from '../ListItem';
import Header from '../Header';
class App extends React.Component<NonNullable<unknown>, StateType> {
  state = {
    count: 2110,
    next: 'https://pokeapi.co/api/v2/item?offset=10&limit=10',
    previous: null,
    results: null,
  };

  componentDidMount = async () => {
    try {
      // const storageData = localStorage.getItem('data');
      const response: Response = await fetch('https://pokeapi.co/api/v2/item');
      const data: StateType = await response.json();
      this.setState(data);
    } catch (error) {
      throw new Error(`error: ${error}`);
    }
  };

  render() {
    return (
      <div className={s.container}>
        <Header />
        <ListItems results={this.state.results} />
      </div>
    );
  }
}

export default App;
