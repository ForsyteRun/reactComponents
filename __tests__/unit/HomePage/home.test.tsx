/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import { default as React } from 'react';
import { renderWithProviders } from '../../../__mocks__/reduxProvide';
import App from '../../../src/App';
import { IFetchData } from '../../../src/types';
import data from './../../../__mocks__/data.json';

global.React = React;
const jsonData: IFetchData = data;

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLoaderData: jest.fn(() => {
    return {
      items: jsonData.items.slice(0, 10),
    };
  }),
}));

describe('Home component', () => {
  let fragment: () => DocumentFragment;
  let rootContainer: HTMLElement;

  beforeEach(() => {
    const { asFragment, container } = renderWithProviders(<App />);

    fragment = asFragment;
    rootContainer = container;
  });

  test('snapshot page when loading', () => {
    expect(fragment()).toMatchSnapshot();
  });

  test('render spinner when loading', () => {
    const spinner = rootContainer.querySelector('.lds-dual-ring');

    expect(spinner).toBeInTheDocument();
  });

  test('render correct title Card', async () => {
    const elements = await screen.findAllByText(/Sex Development/i);
    expect(elements[0]).toBeInTheDocument();
  });

  test('render correct author Card', async () => {
    const elements = await screen.findAllByText(/Zograb/i);
    expect(elements[0]).toBeInTheDocument();
  });

  test('render correct page count in first book', async () => {
    const elements = await screen.findAllByText(/Count/i);

    const nextSiblingValue = elements[0].nextElementSibling?.textContent;
    expect(nextSiblingValue).toEqual('167');
  });

  test('render correct language in first book', async () => {
    const elements = await screen.findAllByText(/language/i);

    const nextSiblingValue = elements[0].nextElementSibling?.textContent;
    expect(nextSiblingValue).toEqual('ru');
  });

  test('render correct img path in first book', async () => {
    const elements: HTMLImageElement[] =
      await screen.findAllByAltText(/Disorders of Sex/i);

    expect(elements[0].alt).toEqual(
      'Disorders of Sex Development in Gynaecology (Russian edition)'
    );
  });
});
