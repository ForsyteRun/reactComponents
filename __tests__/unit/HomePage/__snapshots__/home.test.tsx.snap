// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component snapshot page when loading 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <form
      role="search"
    >
      <input
        name="search"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        search
      </button>
    </form>
    <button>
      get error
    </button>
    <select>
      <option
        value="5"
      >
        5
      </option>
      <option
        selected=""
        value="10"
      >
        10
      </option>
      <option
        value="15"
      >
        15
      </option>
      <option
        value="20"
      >
        20
      </option>
      <option
        value="25"
      >
        25
      </option>
    </select>
    <div
      class="lds-dual-ring"
      style=""
    />
  </div>
</DocumentFragment>
`;
